pc/PlatformDeviceControl.c

```diff
@@ -266,12 +271,35 @@ libusb_device_handle *usbLinkOpen(const char *path)
     }
     usb_free_device(dev);
 #else
+    struct libusb_device_descriptor desc;
+    int res;
+    if ((res = libusb_get_device_descriptor(dev, &desc)) < 0) {
+                printf("Unable to get USB device descriptor: %s\n", libusb_strerror(res));
+        }
+
+    usb_speed_enum = libusb_get_device_speed(dev);
+    const char *speed_str[] = {"Unknown", "Low/1.5Mbps", "Full/12Mbps", "High/480Mbps", "Super/5000Mbps", "Super+/10000Mbps"};
+
     int libusb_rc = libusb_open(dev, &h);
     if (libusb_rc < 0)
     {
         libusb_unref_device(dev);
         return 0;
     }
+    unsigned char sn[XLINK_MAX_MXID];
+    if (libusb_get_string_descriptor_ascii(h, desc.iSerialNumber, sn, sizeof sn) < 0){
```

Comment:
This should be `if (res < 0 || libusb_get_string_descriptor_ascii(h, desc.iSerialNumber, sn, sizeof sn) < 0)`